docs/adr/0005-api-for-spaces.md

```diff
@@ -0,0 +1,118 @@
+
+Most important, the API returns the WebDAV endpoint for each space. With that, clients do not have to make assumptions about WebDAV routes any more.
+
+See [Drive item in Microsoft Graph API](https://docs.microsoft.com/de-de/graph/api/resources/onedrive?view=graph-rest-1.0)
+
+### Get "Home folder"
+
+Retrieve information about the home space of a user. Note: The user has access to more spaces. This call only returns the home space to provide API parity with the Graph API.
+
+API Call: `/me/drive`: Returns the information about the users home folder.
+
+### Get All Spaces of a User
+
+Retrieve a list of available spaces of a user. This includes all spaces the user has access to at that moment, also the home space.
+
+API Call: `/me/drives`: Returns a list of spaces.
+
+### Common Reply
+
+The reply to both calls is either one or a list of [Drive representation objects](https://docs.microsoft.com/de-de/graph/api/resources/drive?view=graph-rest-1.0):
+
+```
+{
+  "id": "string",
+  "createdDateTime": "string (timestamp)",
+  "description": "string",
+  "driveType": "personal | projectSpaces | shares",
+  "driveStatus": "accepted | pending | mandatory",
+  "eTag": "string",
+  "lastModifiedDateTime": "string (timestamp)",
+  "name": "string",
+  "owner": { "@odata.type": "microsoft.graph.identitySet" },
+  "webUrl": "url",
+  "coowner": [ { "@odata.type": "microsoft.graph.identitySet" } ]
+}
+
+```
+
+The meaning of the object are in ownClouds context:
+
+1. **id** - a unique ID identifying the space
+2. **driveType** - describing the type of the space.
+3. **driveStatus** - telling the status
+4. **eTag** - the current ETag of the space
+5. **owner** - an owner object to whom the space belongs
+6. **webUrl** - the relative Webdav path in ownCloud. It must include the `remote.php` part in oC10 installations.
+7. **coowner** - optional array owner objects of the co-owners of a space (*)
+
+The following *driveType* values are available in the first step, but might be enhanced later:
+
+* **personal**: The users home space
+* **projectSpaces**: The project spaces available for the user (*)
+* **shares**: The share jail, contains all shares for the user (*)
+
+The (*) marked types are not defined in the official MS API.
+
+The following *driveStatus* values are available:
+
+* **accepted**: The user has accepted the space and uses it
+* **pending**: The user has not yet accepted the space , but can use it after having it accepted.
+* **mandatory**: This is an mandatory space. Used for the personal- and shares-space. The user can not influence if it is visible or not, it is always available.
+
+### Positive Consequences
+
+- A well understood and mature API from Microsoft adopted to our needs.
+- Prerequisite for Spaces in oCIS.
+- Enables further steps in client development.
+
+### Negative Consequences
+
+- Migration impact on existing installations. Still to be investigated.
+
```

Comment:
```suggestion
- Requires additional webdav endpoint that allows accessing an arbitrary storage space, either
  - with an id: `/dav/spaces/<spaceid>/relative/path/to/file.ext`, or
  - with a global path: `/dav/global/<accessible>/<mount>/<point>/relative/path/to/file.ext`, e.g. `/dav/global/projects/Golive 2021/Resources/slides.odt`

```

Reply:
Yep, but for the old endpoint `/webdav/full/path`, no changes would be needed.